src/globus_sdk/scopes.py

```diff
@@ -116,29 +121,52 @@ def __init__(
         self,
         resource_server: str,
         *,
-        known_scopes: t.Union[t.List[str], str, None] = None,
-        known_url_scopes: t.Union[t.List[str], str, None] = None,
+        known_scopes: ScopeBuilderScopes = None,
+        known_url_scopes: ScopeBuilderScopes = None,
     ) -> None:
         self.resource_server = resource_server
-        self._known_scopes = (
-            list(utils.safe_strseq_iter(known_scopes))
-            if known_scopes is not None
-            else []
-        )
-        self._known_url_scopes = (
-            list(utils.safe_strseq_iter(known_url_scopes))
-            if known_url_scopes is not None
-            else []
-        )
-        self._known_scope_names: t.List[str] = []
-        if self._known_scopes:
-            for scope_name in self._known_scopes:
-                self._known_scope_names.append(scope_name)
-                setattr(self, scope_name, self.urn_scope_string(scope_name))
-        if self._known_url_scopes:
-            for scope_name in self._known_url_scopes:
-                self._known_scope_names.append(scope_name)
-                setattr(self, scope_name, self.url_scope_string(scope_name))
+
+        self._registered_scope_names: t.List[str] = []
+        self._register_scopes(known_scopes, self.urn_scope_string)
+        self._register_scopes(known_url_scopes, self.url_scope_string)
```

Comment:
I really like how small and simple `__init__` became in the course of these changes! 👍

Reply:
I love small methods! Don't make me think about what this thing is doing!